File: task_equivalents/llm_translations/gpt/base/r/data_visualization/02_distribution_boxplot.R
library(readr)
library(dplyr)

college <- read_csv("data/csv/College.csv")

school_name <- college[[1]]
rownames(college) <- school_name
college <- college[, -1]

college$Private <- as.factor(college$Private)

college$Elite <- ifelse(college$Top10perc > 50, "Yes", "No")
college$Elite <- as.factor(college$Elite)

boxplot(Outstate ~ Elite, data = college,
xlab = "Elite", ylab = "Outstate")

par(mfrow = c(2, 2))
hist(college$Apps, breaks = 50, xlab = "new applications", main = "")
hist(college$Enroll, breaks = 45, xlab = "new enrollment", main = "")
hist(college$Expend, breaks = 30, xlab = "Instructional expenditure per student", main = "")
hist(college$Outstate, xlab = "Out-of-state tuition", main = "")

File: task_equivalents/llm_translations/gpt/base/python/data_visualization/02_distribution_boxplot.py
import pandas as pd
import matplotlib.pyplot as plt

college = pd.read_csv("data/csv/College.csv", header=0)
college.index = college.iloc[:, 0]
college = college.iloc[:, 1:]

Elite = pd.Series("No", index=college.index)
Elite[college["Top10perc"] > 50] = "Yes"
college["Elite"] = pd.Categorical(Elite)

plt.figure()
college.boxplot(column="Outstate", by="Elite")
plt.xlabel("Elite")
plt.ylabel("Out-of-state tuition")
plt.title("")
plt.suptitle("")

fig, axes = plt.subplots(2, 2)
axes[0, 0].hist(college["Apps"], bins=50)
axes[0, 0].set_title("# of applications")

axes[0, 1].hist(college["Enroll"], bins=25)
axes[0, 1].set_title("# of new enrollment")

axes[1, 0].hist(college["Expend"], bins=10)
axes[1, 0].set_title("Instructional expenditure per student")

axes[1, 1].hist(college["Outstate"])
axes[1, 1].set_title("Out-of-state tuition")

plt.tight_layout()
plt.show()